runtime/doc/lsp.txt

```diff
@@ -1109,9 +939,259 @@ with({handler}, {override_config})                            *vim.lsp.with()*
+                              this to modify parameters before they are sent.
+      • {on_init?}            (`elem_or_list<vim.lsp.client.on_init_cb>`)
+                              Callback invoked after LSP "initialize", where
+                              `result` is a table of `capabilities` and
+                              anything else the server may send. For example,
+                              clangd sends `initialize_result.offsetEncoding`
+                              if `capabilities.offsetEncoding` was sent to it.
+                              You can only modify the `client.offset_encoding`
+                              here before any notifications are sent.
+      • {on_exit?}            (`elem_or_list<vim.lsp.client.on_exit_cb>`)
+                              Callback invoked on client exit.
+                              • code: exit code of the process
+                              • signal: number describing the signal used to
+                                terminate (if any)
+                              • client_id: client handle
+      • {on_attach?}          (`elem_or_list<vim.lsp.client.on_attach_cb>`)
+                              Callback invoked when client attaches to a
+                              buffer.
+      • {trace?}              (`'off'|'messages'|'verbose'`) Passed directly
+                              to the language server in the initialize
+                              request. Invalid/empty values will (default:
+                              "off")
+      • {flags?}              (`table`) A table with flags for the client. The
+                              current (experimental) flags are:
+                              • allow_incremental_sync (bool, default true):
+                                Allow using incremental sync for buffer edits
+                              • debounce_text_changes (number, default 150):
+                                Debounce didChange notifications to the server
+                                by the given number in milliseconds. No
+                                debounce occurs if nil
+                              • exit_timeout (number|boolean, default false):
+                                Milliseconds to wait for server to exit
+                                cleanly after sending the "shutdown" request
+                                before sending kill -15. If set to false, nvim
+                                exits immediately after sending the "shutdown"
+                                request to the server.
+      • {root_dir?}           (`string`) Directory where the LSP server will
+                              base its workspaceFolders, rootUri, and rootPath
+                              on initialization.
+
+
+
 ==============================================================================
 Lua module: vim.lsp.buf                                              *lsp-buf*
 
+*vim.lsp.CodeActionResultEntry*
```

Comment:
I want to inline this but the issue is that this is used as:

```
---@param results table<integer, vim.lsp.CodeActionResultEntry>
```

So we kind of need to document the class. We could maybe try to inline this documentation a bit differently so the name is preserved?